java/test/com/google/jglome/GlomeTest.java

```diff
@@ -0,0 +1,165 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.jglome;
+
+import static com.google.jglome.TestVector.testVectors;
+import static java.security.spec.NamedParameterSpec.X25519;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.security.InvalidAlgorithmParameterException;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.interfaces.XECPrivateKey;
+import java.security.interfaces.XECPublicKey;
+import java.util.Arrays;
+import java.util.Optional;
+import com.google.jglome.Glome.WrongTagException;
+import org.junit.jupiter.api.Test;
+import com.google.jglome.Glome.CounterOutOfBoundsException;
+import com.google.jglome.Glome.MinPeerTagLengthOutOfBoundsException;
+import com.google.jglome.Glome.StaticSecureRandom;
+
+/**
+ * Contains tests for main.com.google.jglome.Glome functionality.
+ */
+public class GlomeTest {
+
+  final static int N_TEST_VECTORS = 2;
```

Comment:
Could replace all uses with testVectors.length?

Reply:
Or maybe just change a magic constant to this value? what do you think? As it widely used in the code.

Reply:
As in, "N_TEST_VECTORS = testVectors.length;" that should work.